src/compile/data/parse.ts

```diff
@@ -39,17 +39,41 @@ import {TimeUnitNode} from './timeunit';
 import {WindowTransformNode} from './window';
 import {makeWindowFromFacet} from './windowfacet';
 
-function parseRoot(model: Model, sources: Dict<SourceNode>): DataFlowNode {
+export function findSource(data: Data, sources: SourceNode[]) {
+  for (const other of sources) {
+    const otherData = other.data;
+    if (isInlineData(data) && isInlineData(otherData)) {
+      const srcVals = data.values;
+      const otherVals = otherData.values;
+      if (deepEqual(srcVals, otherVals)) {
+        return other;
+      }
+    } else if (isUrlData(data) && isUrlData(otherData)) {
```

Comment:
How about `else if (isUrlData(data) && isUrlData(otherData) && data.url === otherData.url) {` ?

Reply:
Oh, the type narrowing won't work with this I guess.